sign/slhdsa/address.go

```diff
@@ -0,0 +1,92 @@
+package slhdsa
+
+import "encoding/binary"
+
+// See FIPS 205 -- Section 4.2
+// Functions and Addressing
+
+type addrType = uint32
+
+const (
+	addressWotsHash = addrType(iota)
+	addressWotsPk
+	addressTree
+	addressForsTree
+	addressForsRoots
+	addressWotsPrf
+	addressForsPrf
+)
+
+const (
+	addressSizeCompressed    = 22
+	addressSizeNonCompressed = 32
+)
+
+type address struct {
```

Comment:
Might be more efficient with respect to allocations to use [32]byte as address type.